packages/react-dom-bindings/src/client/ReactDOMHostConfig.js

```diff
@@ -1253,7 +1424,11 @@ export function getTextContent(fiber: Fiber): string | null {
 }
 
 export function isHiddenSubtree(fiber: Fiber): boolean {
-  return fiber.tag === HostComponent && fiber.memoizedProps.hidden === true;
+  return (
+    (fiber.tag === HostComponent ||
+      (enableHostSingletons ? fiber.tag === HostSingleton : false)) &&
```

Comment:
This feature is going to be removed but it doesn't really make sense for HostSingletons anyway so you can skip this case.